feat(web-analytics): self managed resources integration

jabahamondes opened the pull request:
## Problem

Need for json upload support

## Changes

This will add support not for json but for any self managed source. User will need to map the respective fields. We create a new logic for marketing analytics. In the future I will moved all the logic in there, for now it just has the main selectors used in web analytics. 

## Did you write or update any docs for this change?

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [x] No docs needed for this change

## How did you test this code?

Manually

Commit:
refactor: Update pipelineNodeNew to use source instead of kind

- Changed the parameter from 'kind' to 'source' in urls.pipelineNodeNew across multiple files.
- Updated related logic in sourceWizardLogic and DataWarehouseIntegrationChoice to accommodate the new parameter.
- Ensured consistency in handling external data sources by using lowercase values for source types.

Commit:
feat(marketing): Add Marketing Analytics configuration and settings

- Introduced  type and integrated it into the  interface.
- Added mock data for  in .
- Updated schema to include  and .
- Created new components for Marketing Analytics settings, including  and .
- Implemented logic for managing source mappings in .
- Added routes and UI elements for Marketing Analytics in the data management scene.
- Established backend support with new models and serializers for marketing analytics configuration.

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update query snapshots

Commit:
fix: Prs comments

Commit:
Merge branch 'feat/adding-marketing-settings-v2' of github.com:PostHog/posthog into feat/adding-marketing-settings-v2

Commit:
fixing snapshot

Commit:
updated snapshot

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

Commit:
refactor: Update NonNativeExternalDataSourceConfiguration to use dataWarehouseSettingsLogic

- Switched data source logic to utilize dataWarehouseSettingsLogic for fetching dataWarehouseSources.
- Maintained existing functionality by keeping sources_map from marketingAnalyticsSettingsLogic.

Commit:
Merge branch 'feat/adding-marketing-settings-v2' of github.com:PostHog/posthog into feat/adding-marketing-settings-v2

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-marketing-settings-v2

Commit:
fix(migrations): correct dependency for 0763_migrate_team_base_currency and add 0764_teammarketinganalyticsconfig migration

Commit:
fix(migrations): update dependency for 0764_teammarketinganalyticsconfig migration to 0763_migrate_team_base_currency

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-marketing-settings-v2

Commit:
feat(migrations): remove obsolete 0763_migrate_team_base_currency migration and add 0763_teammarketinganalyticsconfig model

Commit:
Purpose mistake

Commit:
fixing

Commit:
refactor(schema): move SourceMap type definition to schema-general and update imports

- Moved `SourceMap` type from `marketing-analytics/utils` to `schema-general.ts` for better organization.
- Updated imports in `webAnalyticsLogic.tsx` and `marketingAnalyticsSettingsLogic.ts` to reflect the new location.
- Adjusted the `_sources_map` field in the migration to ensure it defaults to a non-null value.
- Cleaned up comments in `NonNativeExternalDataSourceConfiguration.tsx` for clarity.

Commit:
feat(marketing-analytics): add self-managed external data source configuration and refactor non-native source handling

- Introduced  for user-managed sources.
- Updated  to utilize shared components for better code reuse.
- Enhanced data mapping logic for marketing analytics, including new URL mapping functions and improved table handling.
- Adjusted schema requirements for clicks and impressions in .
- Added new utility functions for source name mapping based on URL patterns.

Commit:
fix(team): update _sources_map field to enforce non-null constraint

- Changed the _sources_map field in TeamMarketingAnalyticsConfig from null=True to null=False to ensure it always has a value.

jabahamondes commented on frontend/src/scenes/data-warehouse/settings/DataWarehouseSourceIcon.tsx:
```diff
@@ -48,6 +48,19 @@ export function mapUrlToProvider(url: string): string {
     return 'BlushingHog'
 }
 
+export function mapUrlToSourceName(url: string): string {
+    if (url.includes('amazonaws.com')) {
+        return 'AWS'
+    } else if (url.startsWith('https://storage.googleapis.com')) {
+        return 'GCS'
+    } else if (url.includes('.blob.')) {
+        return 'Azure'
+    } else if (url.includes('.r2.cloudflarestorage.com')) {
+        return 'Cloudflare'
+    }
+    return 'BlushingHog'
+}
```
It's not the same

Commit:
refactor(data-warehouse): streamline data source URL handling and improve type definitions

- Replaced hardcoded URLs with a base URL constant for better maintainability in `getDataWarehouseSourceUrl`.
- Updated type definitions in `NonNativeExternalDataSourceConfiguration` and `SelfManagedExternalDataSourceConfiguration` for clarity and consistency.
- Adjusted `source_type` in `SharedExternalDataSourceConfiguration` to use a string type for flexibility.
- Changed comment in `removeTableMapping` to reflect the correct behavior of setting field mappings to null instead of undefined.

Commit:
Fixing error when no data in not required

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-marketing-settings-v2

Commit:
Merge branch 'feat/adding-marketing-settings-v2' into feat/adding-s3-and-currency-support-v2

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-s3-and-currency-support-v2

jabahamondes commented on frontend/src/scenes/data-warehouse/settings/DataWarehouseSourceIcon.tsx:
```diff
@@ -48,6 +48,19 @@
 }
 
+export function mapUrlToSourceName(url: string): string {
+    if (url.includes('amazonaws.com')) {
```
Not necessary, we are being consistent with existing logic in mapUrlToProvider

Commit:
feat(web-analytics): update return types to DataWarehouseNode for consistency

Changed return types in webAnalyticsLogic from AnyEntityNode to DataWarehouseNode for improved type safety and clarity in data handling. Updated relevant functions to reflect this change.

Commit:
fix(web-analytics): cast filtered nodes to DataWarehouseNode[]

Added a type assertion to the filter method in webAnalyticsLogic to satisfy the type checker regarding null values. This ensures type safety while maintaining the existing functionality.

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
fix(web-analytics): filter required fields in SharedExternalDataSourceConfiguration

Updated the logic in SharedExternalDataSourceConfiguration to filter required fields from MARKETING_ANALYTICS_SCHEMA, ensuring only necessary fields are considered for mapping. This enhances the configuration validation process.

Commit:
feat(web-analytics): integrate marketing analytics logic into web analytics

Added marketing analytics logic to webAnalyticsLogic, enabling the use of valid external tables for dynamic data warehouse node creation. Updated selectors to filter and map external tables, enhancing the overall data handling and configuration process in the web analytics scene. Refactored related components to utilize the new external table structure for improved consistency and clarity.

Commit:
Merge branch 'feat/adding-s3-and-currency-support-v2' of github.com:PostHog/posthog into feat/adding-s3-and-currency-support-v2

Commit:
Merge branch 'master' into feat/adding-s3-and-currency-support-v2

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
refactor(marketing-analytics): simplify external data source configurations

Removed unused  functions from  and . Updated  to directly render the  within the table configuration, enhancing clarity and reducing redundancy. Ensured  in  defaults to an empty object if  is undefined.

Commit:
Merge branch 'feat/adding-s3-and-currency-support-v2' of github.com:PostHog/posthog into feat/adding-s3-and-currency-support-v2

rafaeelaudibert reviewed (approved):
LGTM, my only concern is that the settings seem to imply this is for revenue analytics - you only figure out it isnt once you read the small text.

<img width="325" alt="image" src="https://github.com/user-attachments/assets/f4e69ed1-0d11-404a-903c-0dab2441f8a2" />

Can we change the headers to make this clearer? Ofc, follow-up PR

Commit:
feat(marketing-settings): add marketing settings to product tree and settings map

- Introduced a new entry for "Marketing settings" in the product tree metadata with a corresponding feature flag.
- Updated the SETTINGS_MAP to include a new section for marketing settings, replacing the previous external data source configurations with a direct reference to the new MarketingAnalyticsSettings component.
- Adjusted the types to reflect the new marketing settings ID.

rafaeelaudibert commented:
Much nicer! I think my settings are the ones that are slightly ugly now 😉. Thank you!

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-s3-and-currency-support-v2

Commit:
Merge branch 'feat/adding-s3-and-currency-support-v2' of github.com:PostHog/posthog into feat/adding-s3-and-currency-support-v2
